ml-agents/mlagents/trainers/settings.py

```diff
@@ -0,0 +1,365 @@
+import attr
+import cattr
+from typing import Dict, Optional, List, Any, DefaultDict, Mapping
+from enum import Enum
+import collections
+import argparse
+
+from mlagents.trainers.cli_utils import StoreConfigFile, DetectDefault, parser
+from mlagents.trainers.cli_utils import load_config
+from mlagents.trainers.exception import TrainerConfigError
+from mlagents.trainers.models import LearningRateSchedule, EncoderType
+
+
```

Comment:
In general these parsing helper methods at the top could use a little love, IMO.  They scatter the parsing / error handling a bit.  Ideally there would be a way to embed them closer to the types/fields themselves.  If that's difficult for some reason at least some comments and/or more verbose field names might be helpful

Reply:
I've moved them into their appropriate classes as static methods - the downside being that certain classes are now quite large. But I think it's an improvement over having those methods up at the top.